src/document-submit.js

```diff
@@ -0,0 +1,111 @@
+/**
```

Comment:
I think we should move this to `src/services` to ensure forbidden deps help avoid bad inclusions.